source/jormungandr/jormungandr/scenarios/distributed.py

```diff
@@ -57,142 +70,145 @@ def _compute_all(future_manager, request, instance, krakens_call):
+                direct_paths_by_mode=context.direct_paths_by_mode,
+                request=request,
+                direct_path_type=StreetNetworkPathType.ENDING_FALLBACK,
+                )
 
         pt_journey_pool = PtJourneyPool(
             future_manager=future_manager,
             instance=instance,
-            requested_orig_obj=requested_orig_obj,
-            requested_dest_obj=requested_dest_obj,
-            streetnetwork_path_pool=streetnetwork_path_pool,
+            requested_orig_obj=context.requested_orig_obj,
+            requested_dest_obj=context.requested_dest_obj,
+            streetnetwork_path_pool=context.streetnetwork_path_pool,
             krakens_call=krakens_call,
-            orig_fallback_durations_pool=orig_fallback_durations_pool,
-            dest_fallback_durations_pool=dest_fallback_durations_pool,
+            orig_fallback_durations_pool=context.orig_fallback_durations_pool,
+            dest_fallback_durations_pool=context.dest_fallback_durations_pool,
             request=request,
-        )
+            )
 
         completed_pt_journeys = wait_and_complete_pt_journey(
             future_manager=future_manager,
-            requested_orig_obj=requested_orig_obj,
-            requested_dest_obj=requested_dest_obj,
+            requested_orig_obj=context.requested_orig_obj,
+            requested_dest_obj=context.requested_dest_obj,
             pt_journey_pool=pt_journey_pool,
-            streetnetwork_path_pool=streetnetwork_path_pool,
-            orig_places_free_access=orig_places_free_access,
-            dest_places_free_access=dest_places_free_access,
-            orig_fallback_durations_pool=orig_fallback_durations_pool,
-            dest_fallback_durations_pool=dest_fallback_durations_pool,
+            streetnetwork_path_pool=context.streetnetwork_path_pool,
+            orig_places_free_access=context.orig_places_free_access,
+            dest_places_free_access=context.dest_places_free_access,
+            orig_fallback_durations_pool=context.orig_fallback_durations_pool,
+            dest_fallback_durations_pool=context.dest_fallback_durations_pool,
             request=request,
-        )
+            )
 
         # At the stage, all types of journeys have been computed thus we build the final result here
         res = []
-
-        for mode in requested_dep_modes:
-            dp = direct_paths_by_mode.get(mode).wait_and_get()
-            if getattr(dp, "journeys", None):
-                if mode == "ridesharing":
-                    switch_back_to_ridesharing(dp, True)
-                res.append(dp)
+        if context.partial_response_is_empty:
+            for mode in requested_dep_modes:
+                dp = context.direct_paths_by_mode.get(mode).wait_and_get()
+                if getattr(dp, "journeys", None):
+                    if mode == "ridesharing":
+                        switch_back_to_ridesharing(dp, True)
+                    res.append(dp)
```

Comment:
First Kraken:
 - journey A to B, we compute from-A and B-to

Second Kraken:
 - journey C to D, we need from-C and D-to, but it seems that we will not compute fallbacks on the second run.